app/controllers/api/news_vote_controller.rb

```diff
@@ -0,0 +1,43 @@
+class Api::NewsVoteController < ApiController
+  def index
+    distribute_reads(max_lag: MAX_ACCEPTABLE_REPLICATION_LAG, lag_failover: true) do
+      news_vote_summary = NewsVote.votes_for_news_item(news_item_id: params[:news_id])
+      if current_user
+        @news_vote = NewsVote.find_by(user: current_user, news_item_id: params[:news_id])
+      end
+
+      if not news_vote_summary
+        return respond_error 'Could not find news item.'
+      end
+
+      respond_success serialize_vote_summary(summary: news_vote_summary, vote: @news_vote)
+    end
+  end
+
+  def create
+    unless current_user
+      respond_error 'Must be logged in to vote.'
+    end
+
+    news_vote = NewsVote.cast_vote(current_user, params[:news_id], params[:direction], current_user.admin?)
+    if not news_vote
+      return respond_error 'Could not save vote.'
+    end
+
+    respond_success serialize_vote_summary(news_vote)
+  end
+
+  private
+
+  def serialize_vote_summary(news_vote = nil, summary: nil, vote: nil)
```

Comment:
Just curious if there's a benefit to overriding this method? If not, I would separate this into two different methods, one to handle `news_vote` and the other for `summary`, i.e.

```
def serialize_news_vote(news_vote)
  news_vote.as_json(only: %i[vote], methods: %i[vote_summary])
end

def serialize_news_summary(summary, vote: nil)
  obj = { vote_summary: summary }
  if vote.present?
    obj[:vote] = vote.vote
  end
  obj.as_json
end
```

Reply:
`serialize_news_vote` may not even be needed